### ISSUE TITLE
minikube start command failed due to deadline exceeded running docker commands

### ISSUE THREAD

**Issue by liranbg** [manusa/actions-setup-minikube#24]
Hey @manusa, Thank you for this useful action!

I've recently noticed some transient errors while installing docker and minikube happening during action execution.
It seems that docker is not ready when minikube command is executed. perhaps a wait until docker is ready before running the command? or retry the minikube start command when a deadline error occurred? 

Attached logs from a failed run:

```
* minikube v1.14.2 on Ubuntu 18.04
  - MINIKUBE_HOME=/home/runner/work/_temp
* minikube 1.15.1 is available! Download it: https://github.com/kubernetes/minikube/releases/tag/v1.15.1
* To disable this notice, run: 'minikube config set WantUpdateNotification false'

* Using the docker driver based on user configuration

X Exiting due to PROVIDER_DOCKER_NOT_RUNNING: deadline exceeded running "docker version --format -": signal: killed
* Suggestion: Restart the Docker service
* Documentation: https://minikube.sigs.k8s.io/docs/drivers/docker/

Error: Command failed:  /home/runner/work/_temp/minikube start --vm-driver=docker --kubernetes-version v1.17.9 --addons registry --addons ingress
    at checkExecSyncError (child_process.js:621:11)
    at Object.execSync (child_process.js:657:15)
    at logExecSync (/home/runner/work/_actions/manusa/actions-setup-minikube/v2.0.1/src/exec.js:8:17)
    at install (/home/runner/work/_actions/manusa/actions-setup-minikube/v2.0.1/src/install.js:24:3)
    at async run (/home/runner/work/_actions/manusa/actions-setup-minikube/v2.0.1/src/index.js:16:3) {
  status: 63,
  signal: null,
  output: [ null, null, null ],
  pid: 5249,
  stdout: null,
  stderr: null
}
```

**Comment by manusa** [comment]
In the more recent versions of the action Docker is no longer being reinstalled/replaced (#16 #17).

Are you using the docker vm-driver for some specific reason?

I'll look into this. I really find it very weird that the GH environment is provided with the Docker daemon not being ready. As you said, retrying this command (`docker version --format -`) a few times before starting Minikube will add some resilience.

**Comment by liranbg** [comment]
Hey @manusa,
We are using docker mainly because it is the cross platform recommented way to use minikube. It allows us all features + room to expand our tests.

The error is transient, and occur rarely but in my opinion, can be avoid with a 1~ minute waiting for docker to be ready or minikube installation.

**Comment by manusa** [comment]
It's really weird that it fails due to Docker not being ready. When running the action with defaults it uses the none/bare-metal driver config which in the ends relies on Docker too and I've never experienced this issue.

Anyway, I'll add a `waitForDocker` method that retries this command (`docker version --format -`) a few times before Minikube is started. But I do feel this check should be done in Minikube itself, the `PROVIDER_DOCKER_NOT_RUNNING` should actually retry a few times before giving up since this is a valid scenario (proven by your issue report).

**Comment by manusa** [comment]
Hi @liranbg, just released [`v2.1.0`](https://github.com/marketplace/actions/setup-minikube-kubernetes-cluster?version=v2.1.0) which will retry and wait every 5 seconds (at most 10 times) the command that Minikube will use to check if Docker is ready.

I'm not sure if this will take care of your problem, but if everything else is fine on the Minikube side, it should.

**Comment by liranbg** [comment]
Thank you very much. seems like it is working for now.

**Comment by manusa** [comment]
Great.
I'm closing the issue now, if the problem reappears feel free to reopen or file a new issue.
